server/core/src/main/java/org/infinispan/server/core/backup/resources/CacheResource.java

```diff
@@ -0,0 +1,280 @@
+         try (OutputStream os = Files.newOutputStream(xmlPath)) {
+            parserRegistry.serialize(os, cacheName, configuration);
+         } catch (XMLStreamException | IOException e) {
+            throw new CacheException(String.format("Unable to create backup file '%s'", xmlFileName), e);
+         }
+      }, "write-cache-config");
+
+      if (cache.isEmpty())
+         return configStage;
+
+      ComponentRegistry cr = cache.getComponentRegistry();
+      ClusterPublisherManager<Object, Object> clusterPublisherManager = cr.getClusterPublisherManager().running();
+      SerializationContextRegistry ctxRegistry = cr.getGlobalComponentRegistry().getComponent(SerializationContextRegistry.class);
+      ImmutableSerializationContext serCtx = ctxRegistry.getPersistenceCtx();
+
+      String dataFileName = dataFile(cacheName);
+      Path datFile = cacheRoot.resolve(dataFileName);
+
+      int bufferSize = configuration.clustering().stateTransfer().chunkSize();
+      Publisher<CacheEntry<Object, Object>> p = clusterPublisherManager.entryPublisher(null, null, null, true,
+            DeliveryGuarantee.EXACTLY_ONCE, bufferSize, PublisherTransformers.identity());
+
+      StorageConfigurationManager scm = cr.getComponent(StorageConfigurationManager.class);
+      boolean keyMarshalling = MediaType.APPLICATION_OBJECT.equals(scm.getKeyStorageMediaType());
+      boolean valueMarshalling = MediaType.APPLICATION_OBJECT.equals(scm.getValueStorageMediaType());
+      PersistenceMarshaller persistenceMarshaller = cr.getPersistenceMarshaller();
+      Marshaller userMarshaller = persistenceMarshaller.getUserMarshaller();
+
+      CompletionStage<Void> contentStage = blockingManager.blockingPublisherToVoidStage(
+            Flowable.using(
+                  () -> Files.newOutputStream(datFile),
+                  output ->
+                        Flowable.fromPublisher(p)
+                              .map(e -> {
+                                 CacheBackupEntry be = new CacheBackupEntry();
+                                 be.key = keyMarshalling ? marshall(e.getKey(), userMarshaller) : (byte[]) scm.getKeyWrapper().unwrap(e.getKey());
+                                 be.value = valueMarshalling ? marshall(e.getValue(), userMarshaller) : (byte[]) scm.getValueWrapper().unwrap(e.getKey());
+                                 be.metadata = marshall(e.getMetadata(), persistenceMarshaller);
+                                 be.internalMetadata = e.getInternalMetadata();
+                                 be.created = e.getCreated();
+                                 be.lastUsed = e.getLastUsed();
+                                 return be;
+                              })
+                              .doOnNext(e -> writeMessageStream(e, serCtx, output))
+                              .doOnError(t -> {
+                                 throw new CacheException("Unable to create cache backup", t);
```

Comment:
Sorry I didn't notice this earlier, but this doesn't do what you want sadly.

You need to do something like https://github.com/infinispan/infinispan/blob/37abd44b44b51eb5b32b23f7f7ddb277f7661021/core/src/main/java/org/infinispan/stream/impl/DistributedCacheStream.java#L378